### DISCUSSION TITLE
Contributors are not updated when commit history was updated

### DISCUSSION THREAD

**Discussion by yellowstonesoftware** [community/community#113236] [Repositories]
### Select Topic Area

Bug

### Body

Hi
Rewrote the commit history after accidentally committing with another user. Unfortunately the Contributors list still shows said user. How can the Contributors be updated to reflect the updated commit history ?
Thanks!

**Comment by stjepanDeveloper** [comment]
Hello @yellowstonesoftware ,

Have you tried using the :
**git commit --amend --author="New Author Name <email@address.com>"**
and 
**git push --force**   
commands ?
These commands should reflect the changes in the commit history. Just be careful if you are using them because they can disrupt the workflow of others in your team.

If this is also not working try checking up on these things :

- ensure that email addresses that are used in the commits are correct since github uses them to link the accounts
- it might take some time to update the contributors list to reflect the new commit history. this delay is usually short though
- check your privacy settings to make sure the commits aren't made with an email address that isn't linked with any github account, because it wont show up the commits
- users choosing not to share their private contributions
- make sure you have pushed the changes to github

Cheers

**Comment by yellowstonesoftware** [comment]
Hi - yes that's exactly what I did, compare: 

https://github.com/yellowstonesoftware/spake2plus4s/commits/main/
and 
https://github.com/yellowstonesoftware/spake2plus4s/graphs/contributors

but yet on the main repository page I see this:
<img width="366" alt="Screenshot 2024-03-19 at 08 52 17" src="https://github.com/community/community/assets/30418911/f43f0e80-6d2d-4407-a18c-721f8f054332">

**Reply by stjepanDeveloper** [reply]
If you used those commands and it's still not updating then it might be connected to Gihub's caching system or the way it tracks contributions. Because that is the most efficient way of doing it that i know of.
I would contact Gihub support in this case then. Maybe theny can manually refresh the repository's contribution data on their end.

**Reply by yellowstonesoftware** [reply]
this was my attempt to contact GitHub support 😂 Guess i'll need to dig deeper to find right contact

**Comment by aaronmedina-dev** [comment]
@yellowstonesoftware 

Were you able to resolve this? 
I have the same issue before and I was able to resolve this by creating a new blank repo, and then migrate.

First is to make sure that the author's traces is completely removed from your old repo
`git log --all --author="email@example.com"`

Create a new blank repo. If you want to use the same repo name, change the name of your old repo first so you can reuse the name.
Next is to migrate.

```
git clone git@github.com:user/old-repo.git
cd old-repo
git remote set-url origin git@github.com:user/new-clean-repo.git
git push --mirror
```
